tests/tests/classes/Service.js

```diff
@@ -0,0 +1,385 @@
+    });
+
+    it('should construct with defaults', () => {
+      const serviceInstance = new Service(S);
+
+      expect(serviceInstance.service).to.be.equal(null);
+      expect(serviceInstance.custom).to.deep.equal({});
+      expect(serviceInstance.plugins).to.deep.equal([]);
+      expect(serviceInstance.functions).to.deep.equal({});
+      expect(serviceInstance.environment).to.deep.equal({});
+      expect(serviceInstance.resources.aws).to.deep.equal({});
+      expect(serviceInstance.resources.azure).to.deep.equal({});
+      expect(serviceInstance.resources.google).to.deep.equal({});
+
+    });
+
+    it('should construct with data', () => {
+      const data = {
+        service: 'testService',
+        custom: {
+          customProp: 'value'
+        },
+        plugins: ['testPlugin'],
+        functions: {
+          functionA: {}
+        },
+        resources: {
+          aws: {
+            resourcesProp: 'value'
+          },
+          azure: {},
+          google: {}
+        }
+      };
+
+      const serviceInstance = new Service(S, data);
+
+      expect(serviceInstance.service).to.be.equal('testService');
+      expect(serviceInstance.custom).to.deep.equal({ customProp: 'value' });
+      expect(serviceInstance.plugins).to.deep.equal(['testPlugin']);
+      expect(serviceInstance.functions).to.deep.equal({ functionA: {} });
+      expect(serviceInstance.environment).to.deep.equal({});
+      expect(serviceInstance.resources.aws).to.deep.equal({ resourcesProp: 'value' });
+      expect(serviceInstance.resources.azure).to.deep.equal({});
+      expect(serviceInstance.resources.google).to.deep.equal({});
+    });
+  });
+
+  describe('#load()', () => {
+    let serviceInstance;
+    before(() => {
+      const SUtils = new Utils();
+      const tmpDirPath = path.join(os.tmpdir(), (new Date).getTime().toString());
+      const serverlessYaml = {
+        service: 'testService',
+        custom: {
+          customProp: 'value'
+        },
+        plugins: ['testPlugin'],
+        functions: {
+          functionA: {}
+        },
+        resources: {
+          aws: {
+            resourcesProp: 'value'
+          },
+          azure: {},
+          google: {}
+        }
+      };
+      const serverlessEnvYaml = {
+        vars: {
+          varA: 'varA'
+        },
+        stages: {
+          dev: {
+            vars: {
+              varB: 'varB'
+            },
+            regions: {
+              aws_useast1: {
+                vars: {
+                  varC: 'varC'
+                }
+              }
+            }
+          }
+        }
+      };
+
+      SUtils.writeFileSync(path.join(tmpDirPath, 'serverless.yaml'), YAML.dump(serverlessYaml));
+      SUtils.writeFileSync(path.join(tmpDirPath, 'serverless.env.yaml'), YAML.dump(serverlessEnvYaml));
+
+      const S = new Serverless({ servicePath: tmpDirPath })
+      serviceInstance = new Service(S);
+    });
+
+    it('should load from filesystem', () => {
```

Comment:
"should load from the filesystem" -> we should make sure that we have full sentences here as it makes it easier to read